runtime-sdk/src/crypto/random.rs

```diff
@@ -1,31 +1,57 @@
+//! Random number generator based on root VRF key and Merlin transcripts.
+use std::cell::RefCell;
+
 use merlin::{Transcript, TranscriptRng};
-use rand_core::{CryptoRng, RngCore};
-use schnorrkel::keys::{ExpansionMode, MiniSecretKey};
+use rand_core::{CryptoRng, OsRng, RngCore};
+use schnorrkel::keys::{ExpansionMode, Keypair, MiniSecretKey};
+
+use oasis_core_runtime::common::crypto::hash::Hash;
 
 use crate::{context::Context, dispatcher, keymanager::KeyManagerError, modules::core::Error};
 
-pub struct Rng {
+/// RNG domain separation context.
+const RNG_CONTEXT: &[u8] = b"oasis-runtime-sdk/crypto: rng";
```

Comment:
:information_source: this is changed to `rng` from `random_bytes`

later changes should probably add a version or something. we'll be relying on that to prevent predictable outputs if we reuse keys from the key manager

Reply:
Good idea, we could start with "rng v2" or similar and just bump if needed.